d2core/d2hero/hero_state.go

```diff
@@ -7,14 +7,16 @@ import (
 
 // HeroState stores the state of the player
 type HeroState struct {
-	HeroName  string                         `json:"heroName"`
-	HeroType  d2enum.Hero                    `json:"heroType"`
-	HeroLevel int                            `json:"heroLevel"`
-	Act       int                            `json:"act"`
-	FilePath  string                         `json:"-"`
-	Equipment d2inventory.CharacterEquipment `json:"equipment"`
-	Stats     *HeroStatsState                `json:"stats"`
-	Skills    map[int]*HeroSkill             `json:"skills"`
-	X         float64                        `json:"x"`
-	Y         float64                        `json:"y"`
+	HeroName   string                         `json:"heroName"`
+	HeroType   d2enum.Hero                    `json:"heroType"`
+	HeroLevel  int                            `json:"heroLevel"`
+	Act        int                            `json:"act"`
+	FilePath   string                         `json:"-"`
+	Equipment  d2inventory.CharacterEquipment `json:"equipment"`
+	Stats      *HeroStatsState                `json:"stats"`
+	Skills     map[int]*HeroSkill             `json:"skills"`
+	X          float64                        `json:"x"`
+	Y          float64                        `json:"y"`
+	LeftSkill  int                            `json:"leftSkill"`
```

Comment:
Could be named LeftSkillID and RightSkillID, although I think its better to change the type to *HeroSkill, and re-use what was previously done for serializing/deserializing those, as it is the same use-case.

Reply:
Thanks @presiyan-ivanov for taking a look! I thought about both of this and I saw the serialization function for the skill, also my forst implementation used exactly that. But I saved a sub json object in the json with the skillPoints. I didn‘t like it because the skillPoints are not needed at this point and so I only used the skillID which made it also a little bit clearer when loading the Skills.
Also I named them LeftSkill and RightSkill because thats what they are called in the PlayerStruct (but yes, they are of type heroSkill, not only a skillID). My heart does not rely on both of these decisions and if you really want, I could change it. Up to you!

Reply:
Good point about the skillPoints being redundant in this scenario. Then maybe just renaming to LeftSkillID & RightSkillID might be best in this case

Reply:
@presiyan-ivanov 
Yes, naming them LeftSkillID & RightSkillID
But now it's merged. Do you think that's worth a new PR or should I just hope to think about it and change it in my next PR?
Also you can change it if you do a PR in the next days...

Reply:
I think its overkill to open a PR just for that, you can do it with another PR, and it is not really that big of a deal if it stays like this anyway :)